### PR TITLE
REST API parameters validation;

### PR DESCRIPTION
**Pull request by yevgenmar**
add function description in swagger;
change dataSource to database in api

Signed-off-by: Yevgeniy Marchenko <yevgeniy.marchenko@ca.ibm.com>

### CODE REVIEW

File: open-metadata-implementation/access-services/analytics-modeling/analytics-modeling-server/src/main/java/org/odpi/openmetadata/accessservices/analyticsmodeling/server/AnalyticsModelingRestServices.java [hunk 1]
```diff
@@ -58,41 +65,48 @@ public AnalyticsModelingOMASAPIResponse getDatabases(String serverName, String u
 		RESTCallToken token = restCallLogger.logRESTCall(serverName, userId, methodName);
 
 		try {
+			validateUrlParameters(serverName, userId, null, null, startFrom, pageSize, methodName);
+			
 			DatabasesResponse response = new DatabasesResponse();
 			List<ResponseContainerDatabase> databases = getHandler()
 					.getDatabaseContextHandler(serverName, userId, methodName).getDatabases(startFrom, pageSize);
```
Comment (davidradl) [inline open-metadata-implementation/access-services/analytics-modeling/analytics-modeling-server/src/main/java/org/odpi/openmetadata/accessservices/analyticsmodeling/server/AnalyticsModelingRestServices.java:72]: I notice that in getDatabases it has 

"
getDatabases(Integer startFrom, Integer pageSize) 
			throws AnalyticsModelingCheckedException {
		
		setContext("getDatabases");

		InstanceProperties instanceProperties = omEntityDao.buildMatchingInstanceProperties(Collections.emptyMap(), true);
		List<EntityDetail> entities = omEntityDao.findEntities(instanceProperties, Constants.DATABASE, 0, 0);
"

This does not look right:
- the supplied startFrom and pagesize is being ignored
- it is hard coding startFrom and pageSize to 0. 0 is a goof default for startFrom but the default pageSize should come from the server config maxPageSize

**Review by davidradl — COMMENTED** [review]
I think change is fine, but as per the comment the getDatabases does not seem to honour the requested paging - I suggest reviewing the other calls are honouring paging.  

You may want a standard method to default startFrom and pageSize if they are not on the  request

File: open-metadata-implementation/access-services/analytics-modeling/analytics-modeling-server/src/main/java/org/odpi/openmetadata/accessservices/analyticsmodeling/server/AnalyticsModelingRestServices.java (hunk 1 above)
Comment (yevgenmar) [reply open-metadata-implementation/access-services/analytics-modeling/analytics-modeling-server/src/main/java/org/odpi/openmetadata/accessservices/analyticsmodeling/server/AnalyticsModelingRestServices.java:72]: This is being refactored using RelationalDataHandler where start index and page size pushed into find function.
In the current implementation, the function retrieved all databases without paging, sorted them by names, and returned page of the sorted set. See last two lines.

**Review by davidradl — APPROVED** [review]
OK it sounds like this is in hand

**Pull request merged**